lib/config/rule-validator.js

```diff
@@ -164,7 +165,10 @@ class RuleValidator {
 
             if (validateRule) {
 
-                validateRule(ruleOptions.slice(1));
+                const slicedOptions = ruleOptions.slice(1);
+                const mergedOptions = deepMergeArrays(rule.meta?.defaultOptions, slicedOptions);
```

Comment:
I think the default options should apply regardless of whether the options are validated (i.e., even if the rule has `schema: false`).

Reply:
I believe they are - and added 299d44fd8 with more tests. Is there a case where they aren't?

Reply:
They apply with inline config comments (with the current implementation), but not if the rule is enabled in the config. For example, when I set `schema: false` in the `no-undef` rule, all tests for the rule fail with `Cannot read properties of undefined (reading 'typeof')`.

Reply:
Thanks!